### ISSUE TITLE
[terra-date-picker] Accessibility Issues - Part 1 of 2

### ISSUE THREAD

**Issue by benbcai** [cerner/terra-framework#1159]
# Bug Report

## Description
The date picker has several accessibility issues that should be addressed or enhanced to increase the accessibility of the component. The issues listed below are less complicated to address than the issues reported on #1158.

- A date in the input is being read as text, not as a date. For example, an input date of `01/15/2020` is being read as "zero one slash fifteen slash twenty twenty". It should be read as "January fifteenth twenty twenty".
    -  [getLocalizedDateForScreenReader](https://github.com/cerner/terra-framework/blob/e7fd6b84a082faad7dc67284e5be2f2285c99421/packages/terra-date-picker/src/react-datepicker/date_utils.js#L380-L393) can be used to set as the aria label on the input. This method can be tweaked to prepend the day of the week or use the `LLLL` format. The returned value would probably needs to be set as the `aria-label` on the input whenever there is a change.

- When navigating within the calendar, provide the day of week in programmatic context for each day. For example, a preselected day of 01/15/2020 is being ready as "January fifteenth twenty twenty". It should be enhanced to read "Wednesday January fifteenth twenty twenty".
    -  [getLocalizedDateForScreenReader](https://github.com/cerner/terra-framework/blob/e7fd6b84a082faad7dc67284e5be2f2285c99421/packages/terra-date-picker/src/react-datepicker/date_utils.js#L380-L393) should be updated to prepend the day of the week or use the `LLLL` format. 

- The accessible name (aria-label) must match the visual label (`label` prop in DatePickerField). If the visual label is “Start Date” then the accessible name must also be “Start Date” – a mismatch in visual and accessible name would prevent users from activating the controls.
    - The `label` prop in DatePickerField should be used as the [aria-label for the input](https://github.com/cerner/terra-framework/blob/e7fd6b84a082faad7dc67284e5be2f2285c99421/packages/terra-date-picker/src/DateInput.jsx#L204) instead of the translated "date" string.

- The color of the Previous and Next month button icons and their background colors have insufficient color contrast.  UX to add a new colors. _{Update: Colors added below in comments}_ 

## Steps to Reproduce

1. Turn on Voice Over. Go to the various [examples](https://engineering.cerner.com/terra-framework/components/terra-date-picker/date-picker/date-picker) and listen to what is announced.

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by pranav300** [comment]
@neilpfeiffer Any update on the color contrast?

**Comment by neilpfeiffer** [comment]
@pranav300 - Let's update the color of the previous and next buttons to the full terra dark/black with opacity.
normal state: `rgba(28, 31, 33, 0.85)`
hover state:  `#1c1f21`

**Comment by scottwilmarth** [comment]
@pranav300 I tested in both JAWS and VoiceOver and both still read the date input as text instead of a date. The dates within the calendar do read correctly as a date though.

**Comment by pranav300** [comment]
> @pranav300 I tested in both JAWS and VoiceOver and both still read the date input as text instead of a date. The dates within the calendar do read correctly as a date though.

Had an offline conversation with @scottwilmarth and resolved it.
